src/ImageProcessor.Web/Caching/DiskCache.cs

```diff
@@ -55,6 +58,161 @@ public class DiskCache : ImageCacheBase
         /// </summary>
         private string virtualCachedFilePath;
 
+        private static object cachePathValidatorLock = new object();
+        private static bool cachePathValidatorCheck;
+
+        /// <summary>
+        /// Stores the resulting validated absolute cache folder path
+        /// </summary>
+        private static string validatedAbsoluteCachePath;
+
+        /// <summary>
+        /// Stores the resulting validated virtual cache folder path - if it's within the web root
+        /// </summary>
+        private static string validatedVirtualCachePath;        
+
+        /// <summary>
+        /// This will get the validated absolute path which is based on the configured value one time
+        /// </summary>
+        /// <param name="originalPath"></param>
+        /// <param name="virtualPath">
+        /// The resulting virtual path if the path is within the webroot
+        /// </param>
+        /// <returns></returns>
+        /// <remarks>
+        /// We are performing this statically in order to avoid any overhead used when performing the validation since
+        /// this occurs for each image when it only needs to be done once
+        /// </remarks>
+        private static string GetValidatedAbsolutePath(string originalPath, out string virtualPath)
+        {
+            var absoluteCachePath = LazyInitializer.EnsureInitialized(
```

Comment:
I didn't know about the static `LazyInitializer` class. This should certainly help with performance.